### ISSUE TITLE
Support IIIF version 3.0 Image API

### ISSUE THREAD

**Issue by azaroth42** [openseadragon/openseadragon#1734]
The IIIF Community is going to publish a new major version of the Image API in a short timeframe. It would be great if OSD supported the new version on day 0. The changes are not going to be difficult from my estimation to simply add in, but some refactoring would additionally make the code easier to understand.

Changes for version 3 that would be great if IIIFTileSource supported:
- [ ] The URI is now in `id`, was `@id` 
- [ ] The `profile` pattern is vastly simplified - now `profile` is a string, and the functionality is in `extraFeatures`
- [ ] The `full` region is now just `max`
- [ ] The `@context` is updated to 3, thus version detection should check for `http://iiif.io/api/image/3/context.json`
- [x] There is a new property `preferredFormats` that allows the publisher to suggest a preferred format for the requests, beyond the default of `jpg`.  (Already implemented)

Refactoring suggestions:
- [ ] Instead of checking for version 1 (`isv1`), having the version as an integer that can be tested against.  This would be beneficial in `configure()`, such that it can be used in the various functions.
- [ ] Perhaps split the code into different tile source classes for different versions, with the protocol test being the first entry point?

**Comment by iangilman** [comment]
@azaroth42 sounds great! Any thoughts on who might be up for doing this work? Is there anyone from the IIIF community who might be interested in getting involved here? 

Looks like @radix-pro, @eroux, @ConnerMan, @bgilbert, @thatcher, @msalsbery, @jpstroop, @acdha, @avandecreme, @hyperknot, @tomcrane, @wilkinson, and @berger-n have all contributed to https://github.com/openseadragon/openseadragon/blob/master/src/iiiftilesource.js ...are any of you interested in pitching in here?

**Comment by lutzhelm** [comment]
Additionally, the [canonical form](https://iiif.io/api/image/3.0/#47-canonical-uri-syntax) of the size parameter is `w,h` instead of `w,` for version 3.0.

I could have a look at this issue.

**Comment by iangilman** [comment]
@lutzhelm That would be lovely! Let me know if you need any info along the way...

**Comment by lutzhelm** [comment]
I'd appreciate any comments on my shot at this in #1764.

I stumbled upon the following things:
* I wanted to be sure my implementations would not break any working code but I found it hard to test solely the functions whose code I changed, so I exposed `canBeTiled()` and `constructLevels()` via a `OpenSeadragon.IIIFTileSource` property `__testonly__`.
* In the following line, `maxLevel` is probably intended to be calculated to include all sizes from 1 pixel up to full width / height, i.e. as logarithm of either height or width to the base of 2. But the second parameter of `Math.log()` will be ignored, so the resulting logarithm is the natural logarithm. There is no immediate need to change that, it just seems very random now. Should I raise an issue? https://github.com/openseadragon/openseadragon/blob/ae4f5867149687440ab3aceac9178762c1bae47d/src/iiiftilesource.js#L132
* I do (again) wonder if the size parameter should be given as `full` instead of `w,` (version 2) / `max` instead of `w,h` (version 3) if the requested size has the same resolution as the full image. This would be especially important for level 0 compliant image services. I can't find my last question on this in the IIIF Slack...
